tests/timeseries/test_moving_window.py

```diff
@@ -104,6 +104,23 @@ async def test_window_size() -> None:
     assert len(window) == 5
 
 
+async def test_wait_for_samples() -> None:
```

Comment:
I think we def. need more tests with different wait parameters

Reply:
[hypothesis](https://hypothesis.readthedocs.io/en/latest/quickstart.html) could be a good match for this, but if it's too much effort to add, then something like `pytest.mark.parametrize` could help:

```py
pytest.mark.parametrize('samples_to_wait_for', [-1, 0, 1, 4, 100])
def test_xxx(samples_to_wait_for: int) -> None:
    ...
```